File: 1.Domain/Domain.Model/PreOfferStage.cs
﻿using Core;
using Domain.Model.Enum;
using System;

namespace Domain.Model
{
    public class PreOfferStage: Entity<int>
    {
        public StageType Type { get; set; }
        public DateTime? Date { get; set; }
        public StageStatus Status { get; set; }
        public int ProcessId { get; set; }
        public Process Process { get; set; }
        public int? UserOwnerId { get; set; }
        public User UserOwner { get; set; }
        public int? UserDelegateId { get; set; }
        public User UserDelegate { get; set; }
        public string RejectionReason { get; set; }
        public int DNI { get; set; } 
        public bool BackgroundCheckDone { get; set; }
        public DateTime? BackgroundCheckDoneDate { get; set; }
        public DateTime? BornDate { get; set; }
        public bool PreocupationalDone { get; set; }
        public DateTime? PreocupationalDoneDate { get; set; }
        public int RemunerationOffer { get; set; }
        public int VacationDays { get; set; }
        public DateTime Firstday { get; set; }
        public string Bonus { get; set; }
    }
}

File: src/entities/pre-offer-stage.ts
import { StageStatusEnum } from './enums/stage-status.enum';
import { SeniorityEnum } from './enums/seniority.enum';
import { HealthInsuranceEnum } from './enums/health-insurance.enum';
import { ReaddressStatus } from './ReaddressStatus';

export class preOfferStage {
  id: number;
  date?: Date;
  dni: number;
  feedback: string;
  status: StageStatusEnum;
  userOwnerId?: number;
  userDelegateId?: number;
  processId: number;
  rejectionReason?: string;
  remunerationOffer: number;
  vacationDays: number;
  healthInsurance: HealthInsuranceEnum
  notes: string;
  firstday: Date;
  bonus: string;
  hireDate: Date;
  seniority: SeniorityEnum;
  backgroundCheckDone: boolean;
  backgroundCheckDoneDate: Date;
  preocupationalDone: boolean;
  preocupationalDoneDate: Date;
  readdressStatus: ReaddressStatus;
}
